File: dart/lib/model/status_pinning_status.dart
part of swagger.api;

class StatusPinningStatus {
  /// The underlying value of this enum member.
  String value;

  StatusPinningStatus._internal(this.value);

  static StatusPinningStatus pinning_ = StatusPinningStatus._internal("pinning");
  static StatusPinningStatus pinned_ = StatusPinningStatus._internal("pinned");
  static StatusPinningStatus failed_ = StatusPinningStatus._internal("failed");
  static StatusPinningStatus queued_ = StatusPinningStatus._internal("queued");
  static StatusPinningStatus offloaded_ = StatusPinningStatus._internal("offloaded");

  StatusPinningStatus.fromJson(dynamic data) {
    switch (data) {
          case "pinning": value = data; break;
          case "pinned": value = data; break;
          case "failed": value = data; break;
          case "queued": value = data; break;
          case "offloaded": value = data; break;
    default: throw('Unknown enum value to decode: $data');
    }
  }

  static dynamic encode(StatusPinningStatus data) {
    return data.value;
  }
}

File: micronaut/src/main/java/io/swagger/model/StatusPinningStatus.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonValue;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * Gets or Sets status.PinningStatus
 */
public enum StatusPinningStatus {
  PINNING("pinning"),
    PINNED("pinned"),
    FAILED("failed"),
    QUEUED("queued"),
    OFFLOADED("offloaded");

  private String value;

  StatusPinningStatus(String value) {
    this.value = value;
  }

  @Override
  @JsonValue
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static StatusPinningStatus fromValue(String text) {
    for (StatusPinningStatus b : StatusPinningStatus.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    return null;
  }
}
